src/backend/tests/it/get_users.rs

```diff
@@ -0,0 +1,186 @@
+    }
+}
+
+#[test]
+fn test_get_users_returns_users() {
+    let pic_setup = setup();
+
+    let users_count = 5;
+    create_users(&pic_setup, 1, users_count);
+
+    let caller = Principal::from_text(CALLER).unwrap();
+
+    let arg = GetUsersRequest {
+        matches_max_length: None,
+        updated_after_timestamp: None,
+    };
+    let get_users_response = query_call::<GetUsersResponse>(&pic_setup, caller, "get_users", arg);
+
+    assert_eq!(
+        get_users_response.expect("Call failed").users.len(),
+        users_count as usize
+    );
+}
+
+#[test]
+fn test_get_users_returns_filtered_users_by_updated() {
+    let pic_setup = setup();
+
+    // Add 10 users
+    let users_count_initial = 10;
+    create_users(&pic_setup, 1, users_count_initial);
+
+    // Add one user that will be updated after the desired timestamp
+    let vc_holder = Principal::from_text(VC_HOLDER).expect("VC Holder principal is invalid");
+
+    let create_profile_response =
+        update_call::<UserProfile>(&pic_setup, vc_holder, "create_user_profile", ());
+    let initial_profile = create_profile_response.expect("Create failed");
+
+    // Advance time before creating more users
+    let (pic, principal) = pic_setup;
+    pic.advance_time(Duration::new(10, 0));
+    let timestamp = pic.get_time();
+    let new_pic_setup = (pic, principal);
+    let timestamp_nanos = timestamp
+        .duration_since(UNIX_EPOCH)
+        .expect("Time went backwards")
+        .as_nanos();
+
+    // Add 10 more users
+    let users_count_after_timestamp = 10;
+    create_users(
+        &new_pic_setup,
+        users_count_initial + 1,
+        users_count_initial + users_count_after_timestamp,
+    );
+
+    // Update one of the users created before timestamp
+    let add_user_cred_arg = AddUserCredentialRequest {
+        credential_jwt: VP_JWT.to_string(),
+        current_user_version: initial_profile.version,
+        credential_spec: CredentialSpec {
+            credential_type: "ProofOfUniqueness".to_string(),
+            arguments: None,
+        },
+        issuer_canister_id: Principal::from_text(ISSUER_CANISTER_ID)
+            .expect("VC Holder principal is invalid"),
+    };
+
+    let _ = update_call::<Result<(), AddUserCredentialError>>(
+        &new_pic_setup,
+        vc_holder,
+        "add_user_credential",
+        add_user_cred_arg.clone(),
+    );
+
+    let caller = Principal::from_text(CALLER).unwrap();
+
+    let arg = GetUsersRequest {
+        matches_max_length: None,
+        updated_after_timestamp: Some(timestamp_nanos as u64),
+    };
+    let get_users_response =
+        query_call::<GetUsersResponse>(&new_pic_setup, caller, "get_users", arg);
+
+    assert_eq!(
+        get_users_response
+            .expect("Call get_users failed")
+            .users
+            .len(),
+        users_count_after_timestamp as usize + 1
+    );
+}
+
+#[test]
+fn test_get_users_returns_requested_users_count() {
```

Comment:
Feature wise are we interested in the "count" value? If yes, we should add a dedicated function as `.count()` on a stable map is less costly than listing, mapping, getting the result and counting. Or that's already the plan?

Reply:
We are not yet interested in the count. But we might be when we implement the summary endpoint. When we come to that, we'll add the new state, I agree.

Reply:
Ok